### PR TITLE
Allow choosing session_id in load_simul and load_for_restart

### LINKED ISSUE #83: Cannot load simulation: KeyError
When I want to load the simulation in the `session_00` directory, I receive the key error which may be related to `info_solver.xml file`.

```python
from snek5000 import load; sim = load()
[11/03/21 09:49:03] WARNING  The .../cbox_run_32x32x8_V1.x1.x1._20 __init__.py:98
                             21-11-02_21-02-30/session_00/info_solver.xml file is missing!                                                                                             
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/.fsnet/data/legi/calcul9/home/khoubani8a/Dev/fluidsim/lib/fluidsim_core/loader.py in import_module_solver(key, entrypoint_grp)
     41     try:
---> 42         solver = solvers[key].load()
     43     except KeyError:

KeyError: 'session'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-1-9bf224db26da> in <module>
----> 1 from snek5000 import load; sim = load()

/.fsnet/data/legi/calcul9/home/khoubani8a/Dev/snek5000/src/snek5000/__init__.py in load_simul(path_dir)
     94     from snek5000.solvers import import_cls_simul, get_solver_short_name
     95 
---> 96     Simul = import_cls_simul(get_solver_short_name(path_dir))
     97 
     98     # Load parameters

/.fsnet/data/legi/calcul9/home/khoubani8a/Dev/fluidsim/lib/fluidsim_core/loader.py in import_cls_simul(key, entrypoint_grp)
     66         key = key[len(entrypoint_grp) + 1:]
     67 
---> 68     solver = import_module_solver(key, entrypoint_grp)
     69     return solver.Simul

/.fsnet/data/legi/calcul9/home/khoubani8a/Dev/fluidsim/lib/fluidsim_core/loader.py in import_module_solver(key, entrypoint_grp)
     42         solver = solvers[key].load()
     43     except KeyError:
---> 44         raise ValueError(
     45             "You have to give a proper solver key. Given: "
     46             f"{key}. Expected one of: {list(solvers)}"

ValueError: You have to give a proper solver key. Given: session. Expected one of: ['kth', 'nek', 'cbox']```

### PR DESCRIPTION
**Pull request by ashwinvis**
Fixes #83

- Add session_id parameter to load_simul
- Add session_id parameter to load_for_restart
- Fail quickly if session directory is accidently provided
- Tests

Maintenance:
- Override Parameters._load_params_simul

### CODE REVIEW

File: tests/test_load_simul.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from snek5000 import load_simul
+
+
+def test_load_simul_error(sim_data):
+    with pytest.raises(ValueError):
```
Comment (paugier) [inline tests/test_load_simul.py:7]: Couldn't we instead load the specified session? i.e. `load_simul(sim_data / "session_00")` equivalent to `load_simul(sim_data, session_id=0)`.

Comment (ashwinvis) [reply tests/test_load_simul.py:7]: It is possible, but I don't think I can support that in an elegant way, without providing some more checks and repeating  code. If the intention is to iterate over session directories, a simple function can achieve that.

```py
def get_sessions_ids(path_dir):
    path_dir = Path(path_dir)
    return (int(session.name.split("_")[-1]) for session in path_dir.glob("session_*"))
```

Comment (paugier) [reply tests/test_load_simul.py:7]: I think one can detect this case with a quite simple check at the beginning of `load_simul`. Something like (maybe slightly more complicated):

```python
if session_id is None and path_dir.name.startswith("session_") and (path_dir.parent / "info_solver.xml").exists():
    session_id = int(path_dir.name[len("session_"):])
    path_dir = path_dir.parent
```

I guess users will tend to use `sim = load()` in the session directories and we should support that.

Comment (ashwinvis) [reply tests/test_load_simul.py]: I implemented this in 0f94728. Please take a look.

Comment (paugier) [reply tests/test_load_simul.py]: Great! I think it's actually elegant! :slightly_smiling_face:

**Pull request merged**